pkg/devfile/devfile.go

```diff
@@ -206,8 +206,8 @@ func CreateDevfileForDockerfileBuild(uri, context string) (data.DevfileData, err
 	return devfileData, nil
 }
 
-// DownloadDevfile downloads devfile from the various possible devfile locations in dir and returns the contents
-func DownloadDevfile(dir string) ([]byte, error) {
+// DownloadDevfile downloads devfile from the various possible devfile locations in dir and returns the contents and its context
+func DownloadDevfile(dir string) ([]byte, string, error) {
 	var devfileBytes []byte
 	var err error
 	validDevfileLocations := []string{Devfile, HiddenDevfile, HiddenDirDevfile, HiddenDirHiddenDevfile}
```

Comment:
looks like the list has a priority. can we have test that a repo contains multiple valid devfiles, and should return the highest priority?

Reply:
updated all tests for this func